File: src/main/java/genst/GenstMountains.java
package genst;

import lotr.common.world.map.LOTRMountains;
import lotr.common.world.map.LOTRWaypoint;
import net.minecraftforge.common.util.EnumHelper;

public class GenstMountains {
	public static void addMountain(String name, double x, double z, float h, int r, int lava) {
		Class<?>[] classArr = {Double.TYPE, Double.TYPE, Float.TYPE, Integer.TYPE, Integer.TYPE};
		Object[] args = {x, z, h, r, lava};
		EnumHelper.addEnum(LOTRMountains.class, name, classArr, args);
	}

	public static void postInit() {
		addMountain("PLACEHOLDER", LOTRWaypoint.TOLFALAS_ISLAND.getX(), LOTRWaypoint.TOLFALAS_ISLAND.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.GREEN_HILLS.getX(), LOTRWaypoint.GREEN_HILLS.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.HIMLING.getX(), LOTRWaypoint.HIMLING.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.TOL_FUIN.getX(), LOTRWaypoint.TOL_FUIN.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.TOL_MORWEN.getX(), LOTRWaypoint.TOL_MORWEN.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.GLADDEN_FIELDS.getX(), LOTRWaypoint.GLADDEN_FIELDS.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.MENELTARMA_SUMMIT.getX(), LOTRWaypoint.MENELTARMA_SUMMIT.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.OATBARTON.getX(), LOTRWaypoint.OATBARTON.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.NEEDLEHOLE.getX(), LOTRWaypoint.NEEDLEHOLE.getY(), 1, 100, 0);
		addMountain("PLACEHOLDER", LOTRWaypoint.NOBOTTLE.getX(), LOTRWaypoint.NOBOTTLE.getY(), 1, 100, 0);
	}
}

File: src/main/java/genst/GenstMountains.kt
package genst

import lotr.common.world.map.LOTRMountains
import lotr.common.world.map.LOTRWaypoint
import net.minecraftforge.common.util.EnumHelper

object GenstMountains {
	private fun addMountain(name: String?, x: Double, z: Double, h: Float, r: Int, lava: Int) {
		val classArr = arrayOf<Class<*>>(
			java.lang.Double.TYPE, java.lang.Double.TYPE, java.lang.Float.TYPE, Integer.TYPE, Integer.TYPE
		)
		val args = arrayOf<Any>(x, z, h, r, lava)
		EnumHelper.addEnum(LOTRMountains::class.java, name, classArr, args)
	}

	fun postInit() {
		addMountain("PLACEHOLDER", LOTRWaypoint.TOLFALAS_ISLAND.x, LOTRWaypoint.TOLFALAS_ISLAND.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.GREEN_HILLS.x, LOTRWaypoint.GREEN_HILLS.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.HIMLING.x, LOTRWaypoint.HIMLING.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.TOL_FUIN.x, LOTRWaypoint.TOL_FUIN.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.TOL_MORWEN.x, LOTRWaypoint.TOL_MORWEN.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.GLADDEN_FIELDS.x, LOTRWaypoint.GLADDEN_FIELDS.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.MENELTARMA_SUMMIT.x, LOTRWaypoint.MENELTARMA_SUMMIT.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.OATBARTON.x, LOTRWaypoint.OATBARTON.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.NEEDLEHOLE.x, LOTRWaypoint.NEEDLEHOLE.y, 1f, 100, 0)
		addMountain("PLACEHOLDER", LOTRWaypoint.NOBOTTLE.x, LOTRWaypoint.NOBOTTLE.y, 1f, 100, 0)
	}
}
